### PR TITLE
Suspend Hue Labs effects when Ambilight is playing

### PR DESCRIPTION
**Pull request by toupeira**
This implements the proposal from https://github.com/zim514/script.service.hue/issues/77

@zim514 thanks for your pointer about the model IDs, I totally missed those! :bow:

Automatically detecting all Hue Labs effects is a great idea, I originally thought about a selection UI too but in this case I don't think we really need any settings.

I ran into some smaller hurdles:

- In my case the virtual toggles have a `modelid: HUELABSVTOGGLE` rather than `HUELABSENUM`, but I guess we can just check for both.
- I only want to disable effects on the lights I have selected for Ambilight in Kodi, others should stay enabled.
- I only want to disable effects if the associated lights are turned on, because enabling an effect will also power on the lights.

This ended up quite convoluted, I have to go through the resourcelinks API to find out which lights/groups are associated which each sensor, the groups API to find out which lights are in which groups, and I can use the `savedLightStates` to check which lights are powered on.

But, it seems to work! I tested a few different scenarios:

- Effects are turned on before playing -> get stopped on play -> get started on pause/stop
- Effects are turned off -> stay turned off on play -> stay turned off on pause/stop
- Lights are turned off -> stay turned off on play -> stay turned off on pause/stop

I'm not handling sensor changes during playback, so the effect will always be resumed to the state it had before playback started. I think this is similar to the existing light state handling, and seems perfectly fine for my usage too.

I also haven't thought yet if/how this shold work with Ambilight disabled, i.e. just using the Play/Pause/Stop scenes. I guess disabling effects could be useful there as well, but will leave that for somebody else to implement :wink:

### CODE REVIEW

File: script.service.hue/resources/lib/AmbiGroup.py [hunk 1]
```diff
@@ -85,8 +87,13 @@ def resumeLightState(self):
             try:
                 self.bridge.lights[L].state(xy=xy, bri=bri, on=on, transitiontime=self.resume_transition)
             except QhueException as exc:
-                xbmc.log("[script.service.hue] onPlaybackStopped: Hue call fail: {}".format(exc))
-                reporting.process_exception(exc)
+                if exc.type_id == 201:  # 201 Param not modifiable because light is off error. 901: internal hue bridge error.
+                    pass
```
Comment (toupeira) [inline script.service.hue/resources/lib/AmbiGroup.py:91]: **note:** I noticed this bug while testing, when the lights are powered off I'd get an error dialog in Kodi when stopping playback. I remember this working fine, but I saw you upgraded Qhue and made some changes to the error handling so maybe this slipped in recently in `main`?

Let me know if this seems correct, also happy to extract this fix into a separate PR!

Comment (zim514) [reply script.service.hue/resources/lib/AmbiGroup.py:91]: Good catch, thanks!

**Comment by zim514** [comment]
Thanks for this, I've merged it but before releasing an update I'll add a setting toggle so that it's only active for those who want it.

**Comment by toupeira** [comment]
@zim514 thanks for merging! Also tested the toggle on latest `main` and it works great :+1:
